src/VisualLinter/Helpers/StylelintHelper.cs

```diff
@@ -0,0 +1,292 @@
+            if (Options.ShouldOverrideStylelintPath)
+            {
+                var overridePath = GetOverrideStylelintPath() ??
+                    throw new Exception("exception: option 'Override Stylelint path' set to true-- but no path is set");
+
+                OutputWindowHelper.DebugLine($"using override stylelint @ {overridePath}");
+
+                return ValidateOverridePath(overridePath);
+            }
+
+            OutputWindowHelper.DebugLine($"UseGlobalStylelint: {Options.UseGlobalStylelint}");
+
+            // resolve global stylelint path
+            if (Options.UseGlobalStylelint)
+            {
+                var globalPath = GetGlobalStylelintPath() ??
+                    throw new Exception("exception: no global stylelint found-- is stylelint installed globally?");
+
+                OutputWindowHelper.DebugLine($"using global stylelint @ {globalPath}");
+
+                return globalPath;
+            }
+
+            // resolve local stylelint path
+            var localPath = GetLocalStylelintPath(directoryPath) ??
+                throw new Exception("exception: no local stylelint found-- is stylelint installed locally?");
+
+            OutputWindowHelper.DebugLine($"using local stylelint @ {localPath}");
+
+            return localPath;
+        }
+
+        internal static string GetIgnorePath(string directoryPath)
+        {
+            OutputWindowHelper.DebugLine($"DisableStylelintIgnore: {Options.DisableStylelintIgnore}");
+
+            // disable stylelint ignore
+            if (Options.DisableStylelintIgnore)
+            {
+                OutputWindowHelper.DebugLine("not using stylelint ignore");
+
+                return null;
+            }
+
+            OutputWindowHelper.DebugLine($"ShouldOverrideStylelintIgnorePath: {Options.ShouldOverrideStylelintIgnorePath}");
+
+            // resolve override stylelint ignore path
+            if (Options.ShouldOverrideStylelintIgnorePath)
+            {
+                var overridePath = GetOverrideStylelintIgnore() ??
+                    throw new Exception("exception: option 'Override Stylelint ignore path' is set to true-- but no path is set");
+
+                OutputWindowHelper.DebugLine($"using override stylelint ignore @ {overridePath}");
+
+                return ValidateOverridePath(overridePath);
+            }
+
+            // resolve stylelint ignore path
+            var solutionPath = VsixHelper.GetSolutionPath() ??
+                throw new Exception("exception: could not get solution path");
+
+            var workingDirectory = new DirectoryInfo(directoryPath);
+
+            var path = FindRecursive(workingDirectory, solutionPath, ResolveIgnorePath);
+
+            OutputWindowHelper.DebugLine($"using stylelint ignore @ {path ?? "not found"}");
+
+            return path;
+        }
+
+        private static string FindRecursive(DirectoryInfo workingDirectory, string end, Func<DirectoryInfo, string> fn)
```

Comment:
This should be a shared method between `EslintHelper.cs` and `StylelintHelper.cs`.